### PR TITLE
[DROOLS-4571] It's possible to set an empty title header

### PR DESCRIPTION
**Pull request by yesamer**
@dupliaka  @gitgabrio @danielezonca Can you please test it and review?

Currently it is possible to set an empty label in header cell type (instance title or property title). This PR adds a check which prevent it. In that case, an error message is thrown.

https://issues.jboss.org/browse/DROOLS-4571

### CODE REVIEW

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/domelements/ScenarioHeaderTextAreaDOMElement.java [hunk 1]
```diff
@@ -44,6 +47,13 @@ public void setScenarioHeaderMetaData(ScenarioHeaderMetaData scenarioHeaderMetaD
 
     @Override
     public void flush(final String value) {
+        if (Objects.isNull(value) || value.isEmpty()) {
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/domelements/ScenarioHeaderTextAreaDOMElement.java]: Shouldn't we rather call `value.trim().isEmpty()` ?

Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/domelements/ScenarioHeaderTextAreaDOMElement.java]: @jomarko You're right. Done.

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/domelements/ScenarioHeaderTextAreaDOMElementTest.java [hunk 2]
```diff
@@ -72,6 +74,20 @@ public void flushMetadataNotEqualsValue() {
...
+        verify(scenarioHeaderTextAreaDOMElement, never()).internalFlush(any());
+    }
+
+    @Test
+    public void flushEmptyString() {
+        scenarioHeaderTextAreaDOMElement.flush("");
+        verify(eventBusMock, times(1)).fireEvent(isA(ScenarioNotificationEvent.class));
+        verify(scenarioHeaderTextAreaDOMElement, never()).internalFlush(any());
+    }
+
```
Comment (gitgabrio) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/domelements/ScenarioHeaderTextAreaDOMElementTest.java]: @yesamer 
Could you please add a test with a string containing only spaces ?

Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/domelements/ScenarioHeaderTextAreaDOMElementTest.java]: @gitgabrio Done!

**Comment by yesamer** [comment]
jenkins retest this please

**Comment by yesamer** [comment]
jenkins retest this please

**Comment by yesamer** [comment]
jenkins retest this please

**Pull request merged**